index.bs

```diff
@@ -498,7 +498,70 @@ Add the step:
 
 # Fenced Frame monkeypatches # {#fenced-frame-monkeypatches}
 
-Issue: Modify [=send a beacon=] when the [=fetch=] integration is specified.
+Modify {{Fence/reportEvent()}} as follows:
+
+Replace the step
+
+> Run <a spec=fenced-frame>report an event</a> using <var ignore>instance</var>'s
+> [=fenced frame config instance/fenced frame reporter=] with |destination|,
+> |event|'s {{FenceEvent/eventType}}, and |event|'s {{FenceEvent/eventData}}.
+
+with
+
+Run <a spec=fenced-frame>report an event</a> using <var ignore>instance</var>'s
+[=fenced frame config instance/fenced frame reporter=] with |destination|,
+|event|'s {{FenceEvent/eventType}}, |event|'s {{FenceEvent/eventData}},
+and [=this=]'s [=relevant global object=]'s [=associated Document=].
+
+Modify <a spec=fenced-frame>report an event</a> as follows:
+
+Add a [=Document=] parameter called |document|.
+
+Replace the step
+
+> <a spec=fenced-frame>Send a beacon</a> with <var ignore>reporting map</var>[|destination|],
+> |eventType|, and |eventData|.
+
+with
+
+<a spec=fenced-frame>Send a beacon</a> with <var ignore>reporting map</var>[|destination|],
+|eventType|, |eventData|, and |document|.
+
+Modify [=finalize a reporting destination=] as follows:
```

Comment:
I think [this algorithm](https://wicg.github.io/fenced-frame/#finalize-a-reporting-destination) is meant to be called from in parallel, so we shouldn't really have direct access to a Document object here. @gtanzer is much more familiar with this part of the config-generating API integration though so let me loop him in here.

Reply:
The implementation doesn't take a document parameter, so I'm not sure why the spec would need to. But I haven't looked into the details yet.
https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:content/browser/fenced_frame/fenced_frame_reporter.h;drc=3e67c0491457f1a05fb7816346fc6d9676bb7a16;bpv=1;bpt=1;l=124?q=fenced_frame_reporter&ss=chromium

Reply:
Updated to add the permission policy value and the context origin to pending event for later use.

Reply:
The context origin is the top-level origin (https://source.chromium.org/chromium/chromium/src/+/main:content/browser/attribution_reporting/attribution_host.cc;l=444).

I think plumbing the permissions policy value matches the implementation as well, which indicates whether the feature is allowed in the initiator document (https://source.chromium.org/chromium/chromium/src/+/main:content/browser/attribution_reporting/attribution_host.cc;l=436).

Reply:
Great, OK that makes sense!

Reply:
Thanks for the reviews!